Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/HDRColorGrading.azsl

```diff
@@ -0,0 +1,212 @@
+
+#include <Atom/Features/SrgSemantics.azsli>
+
+#include <Atom/Features/PostProcessing/FullscreenPixelInfo.azsli>
+#include <Atom/Features/PostProcessing/FullscreenVertex.azsli>
+
+#include <Atom/Features/ColorManagement/TransformColor.azsli>
+#include <Atom/Features/PostProcessing/AcesColorSpaceConversion.azsli>
+#include <3rdParty/Features/PostProcessing/PSstyleColorBlends_Separable.azsli>
+#include <3rdParty/Features/PostProcessing/PSstyleColorBlends_NonSeparable.azsli>
+#include <3rdParty/Features/PostProcessing/KelvinToRGB.azsli>
+
+#define FLOAT_EPSILON     1.192092896e-07 // 1.0 + FLOAT_EPSILON != 1.0, smallest positive float
+#define FLOAT_MIN         FLOAT_32_MIN // Min float number that is positive
+#define FLOAT_MAX         FLOAT_32_MAX // Max float number representable
+
+#define saturate(value) clamp(value, FLOAT_EPSILON, 1.0f)
+
+// ACES constants
+static const float ACEScc_MIDGREY = 0.4135884;  // is correct
+
+ShaderResourceGroup PassSrg : SRG_PerPass_WithFallback
+{
+    // get the framebuffer
+    Texture2D<float4> m_framebuffer;
+
+    // framebuffer sampler
+    Sampler LinearSampler
+    {
+        MinFilter = Linear;
+        MagFilter = Linear;
+        MipFilter = Linear;
+        AddressU = Clamp;
+        AddressV = Clamp;
+        AddressW = Clamp;
+    };
+
+    float m_colorGradingExposure;
+    float m_colorGradingContrast;
+    float m_colorGradingHueShift;
+    float m_colorGradingPreSaturation;
+    float m_colorFilterIntensity;
+    float m_colorFilterMultiply;
+    float m_kelvin;
+    float m_kelvinLumPreservation;
+    float m_kelvinColorMix;
+    float m_whiteBalance;
+    float m_whiteBalanceTint;
+    float m_splitToneBalance;
+    float m_splitToneMix;
+    float m_colorGradingPostSaturation;
+    float m_smhShadowsStart;
+    float m_smhShadowsEnd;
+    float m_smhHighlightsStart;
+    float m_smhHighlightsEnd;
+    float m_smhMix;
+
+    float3 m_channelMixingRed;
+    float3 m_channelMixingGreen;
+    float3 m_channelMixingBlue;
+
+    float4 m_colorFilterSwatch;
+    float4 m_splitToneShadowsColor;
+    float4 m_splitToneHighlightsColor;
+
+    float4 m_smhShadowsColor;
+    float4 m_smhMidtonesColor;
+    float4 m_smhHighlightsColor;
+
+    // my color grading output
+    float4 m_color;
+}
+
+// Below are the color grading functions. These expect the frame color to be in ACEScg space.
+// If a color parameter is introduced from external sources (e.g. Editor color picker which uses Linear RGB), 
+// it will be converted into ACEScg as needed.
+float3 ColorGradePostExposure (float3 frame_color, float exposure) 
+{
+    frame_color *= pow(2.0f, exposure);
+	return frame_color;
+}
+
+// The contrast equation is performed in ACEScc (logarithmic) color space.
+float3 ColorGradingContrast (float3 frame_color, float midgrey, float amount) 
+{
+    float contrast_adjustment = amount * 0.01f + 1.0f;
+    frame_color = TransformColor(frame_color.rgb, ColorSpaceId::ACEScg, ColorSpaceId::ACEScc);
```

Comment:
Why do we want to do all the work in ACEScc eventually? Aren't some operations better done in ACEScg space?

Reply:
ACEScc is made for color corrections - basically everything we're doing in this file. So it'd be much cheaper if we could go form ACEScg -> ACEScc, do all color corrections, then go ACEScc -> ACEScg at the end. That keeps the number of (expensive) color conversions down to just 2.

Reply:
weirdly the ACEScc toe is in the negative range with the ACEScg value of 0 getting transformed to -0.3013698630. I'm not sure why it works that way. We'll have to be careful about clamps to 0 in the ACEScc space since it'll basically just clamp the entire toe. 🤷‍♂️